src/main.py

```
@@ -382,6 +408,26 @@ async def handle_mcp(request, env) -> Response:
 # ---------------------------------------------------------------------------
 # Internal helpers used by both direct API and MCP
 # ---------------------------------------------------------------------------
+def _detect_injection(text: str) -> tuple[bool, str]:
+  """ Defense-in-depth: strip high-confidence prompt-injection fragements while preserving legitimate surrounding user intent"""
+  cleaned = text or ""
+  detected = False
+
+for pattern in INJECTION_PATTERNS:
+  cleaned, match_count = pattern.subn(" ", cleaned)
```

Ruff (0.15.9)

[error] Undefined name `cleaned`

(F821)